### PR TITLE
local file path features

### PR DESCRIPTION
**Pull request by joshspicer**
closes https://github.com/devcontainers/cli/issues/136

Updates reference implementation to conform to the spec as it related to 'local file path' features.

Spec ref: https://containers.dev/implementors/features-distribution/#addendum-locally-referenced

- [x] Tests
- [x] fix `devcontainer feature test` (uses absolute paths to generate configs)

### CODE REVIEW

File: src/spec-node/featuresCLI/testCommandImpl.ts [hunk 1]
```diff
@@ -300,9 +301,16 @@ async function generateDefaultProjectFromFeatures(
...
 	const features = featuresToTest
-		.map((x) => `"${collectionsDirectory}/src/${x}": {}`)
+		.map((x) => `"./${x}": {}`)
 		.join(',\n');
 
+	for (const featureId of featuresToTest) {
+		// Copy the feature source code to the temp folder
+		const pathToFeatureSource = `${collectionsDirectory}/src/${featureId}`;
+		await cpDirectoryLocal(pathToFeatureSource, `${tmpFolder}/.devcontainer/${featureId}`);
+		process.stdout.write('AFTER\n');
```
Comment (chrmarti) [inline src/spec-node/featuresCLI/testCommandImpl.ts:311]: Debugging output that should be removed before merging?

File: src/test/container-features/configs/dockerfile-with-v2-local-features-config-inside-dev-container-folder/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+This is a README in a project.
```
Comment (chrmarti) [inline src/test/container-features/configs/dockerfile-with-v2-local-features-config-inside-dev-container-folder/README.md:1]: Do we need that?

File: src/test/container-features/configs/invalid-configs/dockerfile-with-v2-local-features-no-dev-container-folder/local-features/localFeatureA/install.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+echo "Activating feature 'localFeatureA'"
+
+GREETING=${GREETING:-undefined}
+echo "The provided greeting is: $GREETING"
+
+tee /usr/hello.sh > /dev/null \
```
Comment (chrmarti) [inline src/test/container-features/configs/invalid-configs/dockerfile-with-v2-local-features-no-dev-container-folder/local-features/localFeatureA/install.sh:8]: See feature-template for simplified version. I suggest to copy that around for clarity.

File: src/test/container-features/configs/invalid-configs/dockerfile-with-v2-local-features-no-dev-container-folder/local-features/localFeatureB/install.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+echo "Activating feature 'localFeatureB'"
+
+echo "The provided favorite color is: $FAVORITE"
+
+tee /usr/color.sh > /dev/null \
```
Comment (chrmarti) [inline src/test/container-features/configs/invalid-configs/dockerfile-with-v2-local-features-no-dev-container-folder/local-features/localFeatureB/install.sh:7]: Also: See feature-template for simplified version. I suggest to copy that around for clarity.

**Review by chrmarti — CHANGES_REQUESTED** [review]

File: src/spec-node/featuresCLI/testCommandImpl.ts (hunk 1 above)
Comment (joshspicer) [reply src/spec-node/featuresCLI/testCommandImpl.ts:311]: Ah, thought i'd removed this. Thanks

**Review by chrmarti — APPROVED** [review]

**Review by chrmarti — APPROVED** [review]

**Pull request merged**